### DISCUSSION TITLE
Nouns Proposal/Voting system

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#412] [Frontend] [answered]
How did you build the proposal and voting system on nouns.wtf? Is it all through the Alchemy API?

**Comment by solimander** [comment, marked as answer]
Yes, we use `useDapp` to wrap contract interactions and Alchemy as our JSON-RPC provider. In addition, we use The Graph for fetching proposal information to reduce the number of chain requests, but all voting is handled over JSON-RPC.
